### PR TITLE
Feature/cli arguments

### PR DESCRIPTION
**Pull request by sebschmitt**

### CODE REVIEW

File: include/argparser.h [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                std::istringstream in(&s);
+                in >> vt;
+                t.push_back(vt);
+            }
+
+            return t;
+        }
+
+        std::string value;
+        bool _found{false};
```
Comment (JanMaier97) [inline include/argparser.h:48]: Which convention will we use for private variables? You sometimes use an underline as prefix.

File: src/argparser.cpp [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        if (argc == 1) return;
+
+
+        std::string all_args;
+        for (int i = 1; i < argc; i++) {
+            all_args += std::string(argv[i]);
+            all_args += " ";
+        }
+
+        std::vector<std::string> all_names;
```
Comment (JanMaier97) [inline src/argparser.cpp:74]: In .cpp files you can you can use `using namespace std` below the import statements, in order to reduce the verbosity and improve readability.

File: src/main.cpp [hunk 3]
```diff
@@ -8,35 +8,80 @@
...
+    }
 
-    filesystem::path calibrationFile("./resources/calibration.xml");
-    filesystem::path path("./resources/chessboard_images");
+    if (a_loadCalibration.isFound()) {
+        calb.loadCalibration(filesystem::path(a_loadCalibration.getValue<string>()));
+    }
 
-    if (!filesystem::exists(calibrationFile)) {
+    if (a_calibrationImages.isFound()) {
```
Comment (JanMaier97) [inline src/main.cpp:57]: What happens, when loadCalibration and calibrationImages are specified? Right now the file would be loaded and then regenerated.

File: src/main.cpp [hunk 4]
```diff
@@ -8,35 +8,80 @@
...
+        unsigned int cornersColumn = a_calibrationCalibrateColumn.getValue<int>();
 
         calb.calibrate(filepaths, cv::Size(cornersColumn, cornersRow));
-        calb.saveCalibration(calibrationFile);
-    } else {
-        calb.loadCalibration(calibrationFile);
     }
 
-    /* calb.loadCalibration("./resources/calibration.xml"); */
+    if (a_saveCalibration.isFound()) {
```
Comment (JanMaier97) [inline src/main.cpp:82]: Maybe check if the calibration is loaded from a file. If it is the same file saving isn't necessary.

**Review by JanMaier97 — APPROVED** [review]

File: src/argparser.cpp (hunk 2 above)
Comment (sebschmitt) [reply src/argparser.cpp:74]: I know. But I didn't want to use this to make clear which namespace we are using here.

File: src/main.cpp (hunk 3 above)
Comment (sebschmitt) [reply src/main.cpp:57]: This is correct. The question is, what behaviour do we except? An error message?

File: include/argparser.h (hunk 1 above)
Comment (sebschmitt) [reply include/argparser.h:48]: Good question

File: src/main.cpp (hunk 4 above)
Comment (sebschmitt) [reply src/main.cpp:82]: See above

File: src/main.cpp (hunk 3 above)
Comment (JanMaier97) [reply src/main.cpp:57]: An error message and exiting the program are enough.

File: src/argparser.cpp (hunk 2 above)
Comment (JanMaier97) [reply src/argparser.cpp:74]: Okey, that's fine for me.

File: include/argparser.h (hunk 1 above)
Comment (JanMaier97) [reply include/argparser.h:48]: I don't have a strong opinion about this, but I would choose no prefix. Then we do not need to refactor the code we have until now.

**Pull request merged**